tst/unit_test.cpp

```diff
@@ -0,0 +1,39 @@
+//
+// Created by kirill on 22.04.2020.
+//
+
+#include <gtest/gtest.h>
+#include "unit.h"
+
+TEST(unit,constructor_test)
+{
+    {
+        Unit unit;
+        EXPECT_EQ(unit.Data().size(), 0);
+    }
+
+    {
+        Unit unit(1000);
+        EXPECT_EQ(unit.Data().size(),1000);
+    }
+
+    {
+        Unit unit(-10);
+        EXPECT_EQ(unit.Data().size(),0);
+    }
+
+    {
+        Unit unit(0);
+        EXPECT_EQ(unit.Data().size(),0);
+    }
+}
+
+TEST(unit,refill_test)
+{
+    Unit unit(20);
```

Comment:
This test case does not test refill method
To test it you need firstly fill the unit with some data (not zero data) and then do refill
And if in this case your unit will consist of zeros you can say that this test really test the method